src/statedb/database_map.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef DATABASE_MAP_HPP
+#define DATABASE_MAP_HPP
+
+#include <vector>
+#include "goldilocks_base_field.hpp"
+#include <nlohmann/json.hpp>
+#include <mutex>
+
+using namespace std;
+using json = nlohmann::json;
+
+class DatabaseMap;
+
+
+class DatabaseMap
+{
+public:
+    typedef map<string, vector<Goldilocks::Element>> MTMap;
```

Comment:
if you don't need MTMap and ProgramMap to be sorted, you could declare them as std::unordered_map which is better in terms of performance because it uses a hash and usually has a constant cost.